pkg/egressgateway/net.go

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package egressgateway
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/linux_defaults"
+	"github.com/cilium/cilium/pkg/datapath/linux/route"
+)
+
+var zeroIPv4Net = &net.IPNet{IP: net.ParseIP("0.0.0.0"), Mask: net.CIDRMask(0, 32)}
+
+func getIfaceFirstIPv4Address(ifaceName string) (net.IPNet, int, error) {
+	dev, err := netlink.LinkByName(ifaceName)
+	if err != nil {
+		return net.IPNet{}, 0, err
+	}
+
+	addrs, err := netlink.AddrList(dev, netlink.FAMILY_ALL)
```

Comment:
Would we benefit from

```suggestion
	addrs, err := netlink.AddrList(dev, netlink.FAMILY_V4)
```